class/defaults.yml

```diff
@@ -1,16 +1,17 @@
 parameters:
+  multi_instance: true
   keycloak:
     namespace: syn-keycloak
-    release_name: keycloak
+    release_name: keycloak-${_instance}
     charts:
       keycloak: '10.3.1'
     # FQDN should be overwritten on the cluster level
     fqdn: keycloak.example.com
     # Keycloak Admin
     admin:
-      secretname: keycloak-admin-user
+      secretname: ${keycloak:release_name}-admin-user
```

Comment:
probably worth to pick whether you use `${keycloak:release_name}` or `${_instance}` throughout the component, especially if you move to having the release_name be the instance name without any prefix. Currently there's a bit of a mix.

Reply:
With `release_name = _instance` as suggested it doesn't matter really.
However, if someone overrides `release_name` with a different suffix or prefix, some resources won't be matching expected names. I'll double check which ones are relevant...

Reply:
So, testing locally with
```
  keycloak:
    database:
      secretname: ${_instance}-postgresql # if this would be the default value
    release_name: my-release # custom value
```
resulted in secret references that would not be existing resp. mountable in an actual statefulset for the builtin database (not without customizing `database.secretname` as well, unit test confirmed). With `secretname: ${keycloak:release_name}-postgresql` as default I would only need to customize 1 parameter: `release_name` itself.

WDYT?

Reply:
Wouldn't it be a possibility just to use different namespace / app? Would prevent of having to prefix all the objects?

Reply:
>  With secretname: ${keycloak:release_name}-postgresql as default I would only need to customize 1 instead of 3 parameter: just release_name itself.

I don't mind either option, I would just like to have it as consistent as possible.

> Wouldn't it be a possibility just to use different namespace / app? Would prevent of having to prefix all the objects?

In general it's up to the component writer to decide how to implement multiple instances. Requiring each instance to have a separate namespace is a completely valid choice. I think @ccremer will implement this approach for now, so we could consider dropping the prefixes again.

For some other components we allow instances to share a namespace, so each instance's objects must be distinguishable by their names.

Reply:
We discovered that ArgoCD's App spec is problematic as a single app cannot deploy to multiple namespaces if the `metadata.namespace` is missing from the manifests. The `spec.destination.namespace` configures the default for "namespace-less" manifests. However, since we're using Helm charts, setting a namespace to the template-rendering step is not exactly easy, so it's fine for us to define an ArgoCD app per instance, in which each App sets the namespace.

So yes, removing the prefixes again could be done. After all, they are just defaults, if a user absolutely needs multiple instances in the same namespace somehow, they need to override them where necessary.

Reply:
I've changed it to fixed `release_name` and `secretname` now. Those are the only 3 parameters I could find where the name could matter, the rest is rendered through helm anyway.